packages/suite-desktop-core/e2e/support/pageObjects/trading/tradingPage.ts

```diff
@@ -257,24 +317,78 @@ export class TradingPage {
         await this.paymentMethodOption(method).click();
     }
 
+    @step()
+    async selectSuiteReceiveAccount(index: number) {
+        await expect(this.receiveAddressPicker).toBeVisible();
+        await this.receiveAddressPicker.click();
+
+        await expect(this.receiveAccountModal).toBeVisible();
+        const suiteOptionsCount = await this.receiveAccountModalSuiteOption.count();
+        expect.soft(suiteOptionsCount).toBeGreaterThan(0);
+        await this.receiveAccountModalSuiteOption.nth(index).click();
+        await expect(this.receiveAccountModal).toBeHidden();
+    }
+
+    @step()
+    async selectNonSuiteReceiveAccount(receiveAddress: string, extraField?: string) {
+        await expect(this.receiveAddressPicker).toBeVisible();
+        await this.receiveAddressPicker.click();
+
+        await expect(this.receiveAccountModal).toBeVisible();
+        const nonSuiteOptionsCount = await this.receiveAccountModalNonSuiteOption.count();
+        expect.soft(nonSuiteOptionsCount).toEqual(1);
+        await this.receiveAccountModalNonSuiteOption.nth(0).click();
+        await expect(this.receiveAddressModal).toBeVisible();
+        await expect(this.receiveAddressInput).toBeVisible();
+        await this.receiveAddressInput.fill(receiveAddress);
+
+        if (extraField) {
+            await expect(this.extraFieldSwitch).toBeVisible();
+            await expect(this.extraFieldSwitch).not.toBeChecked();
+            await this.extraFieldSwitch.check();
+            await expect(this.extraFieldInput).toBeVisible();
+            await this.extraFieldInput.fill(extraField);
+        }
+
+        await expect(this.receiveAddressModalConfirmButton).toBeVisible();
+        await expect(this.receiveAddressModalConfirmButton).toBeEnabled();
+        await this.receiveAddressModalConfirmButton.click();
+    }
+
+    @step()
+    async selectAddSuiteReceiveAccount(index: number, discoveryCallback: () => Promise<void>) {
+        await expect(this.receiveAddressPicker).toBeVisible();
+        await this.receiveAddressPicker.click();
+
+        await expect(this.receiveAccountModal).toBeVisible();
+        const addSuiteOptionsCount = await this.receiveAccountModalAddSuiteOption.count();
+        expect.soft(addSuiteOptionsCount).toEqual(1);
+        await this.receiveAccountModalAddSuiteOption.nth(0).click();
+
+        await expect(this.addAccountButton).toBeVisible();
+        await this.addAccountButton.click();
+
+        await discoveryCallback();
```

Comment:
discoveryShouldFinish() is method on our page object that is available intencionally everywhere.
We could do 
```suggestion
    async selectAddSuiteReceiveAccount(index: number,) {
        await expect(this.receiveAddressPicker).toBeVisible();
        await this.receiveAddressPicker.click();

        await expect(this.receiveAccountModal).toBeVisible();
        const addSuiteOptionsCount = await this.receiveAccountModalAddSuiteOption.count();
        expect.soft(addSuiteOptionsCount).toEqual(1);
        await this.receiveAccountModalAddSuiteOption.nth(0).click();

        await expect(this.addAccountButton).toBeVisible();
        await this.addAccountButton.click();
        await this.page.discoveryShouldFinish() 
```
and get rid of passing the discovery method thru params. Unless there is a purpose for that.

Reply:
done